Pyarrow 20.0.0 breaks parquet tests

Why do the tests fail? These tests sporadically fail locally, but not always. And I get these errors also sporadically on master.

_Originally posted by @siemdejong in https://github.com/fsspec/filesystem_spec/issues/1829#issuecomment-2834139059_
            
Tests related to parquet fail, e.g.
https://github.com/fsspec/filesystem_spec/actions/runs/14701863057/job/41252773151?pr=1829
```python
>               assert bad_key not in data
E               assert (4032, 4040) not in {(0, 64): b'PAR1\x15\x04\x15P\x15<L\x15\n\x15\x00\x12\x00\x00(\x00\x00\r\x01\x00\x02\r\x08\x00\x04\r\x08<\x01\x00\x00\x00\x00\x00\x00\x00\x03\x00\x00\x00\x00\x00\x00\x00\x15\x00\x15.\x152,\x15P\x15\x10\x15\x06\x15\x06\x1c', (64, 128): b'\x18\x08\x04\x00\x00\x00\x00\x00\x00\x00\x18\x08\x00\x00\x00\x00\x00\x00\x00\x00\x16\x00(\x08\x04\x00\x00\x00\x00\x00\x00\x00\x18\x08\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x17X\x02\x00\x00\x00P\x01\x03\x0b\x88FD#\xa2\x11\xd1\x88h', (128, 192): b'D4"\x1a\x11\x8d\x15\x04\x15\x80\x05\x15\xd6\x02L\x15P\x15\x00\x12\x00\x00\xc0\x02\x00\x00\r\x01\x00\x01\r\x08\x00\x02\r\x08\x00\x03\r\x08\x00\x04\r\x08\x00\x05\r\x08\x00\x06\r\x08\x00\x07\r\x08\x00\x08\r\x08\x00\t\r\x08', (192, 256): b'\x00\n\r\x08\x00\x0b\r\x08\x00\x0c\r\x08\x00\r\r\x08\x00\x0e\r\x08\x00\x0f\r\x08\x00\x10\r\x08\x00\x11\r\x08\x00\x12\r\x08\x00\x13\r\x08\x00\x14\r\x08\x00\x15\r\x08\x00\x16\r\x08\x00\x17\r\x08\x00\x18\r\x08\x00\x19\r\x08', ...}
```

Can reproduce locally with `pyarrow` version 20.0.0. Downgrading `pyarrow` to 19.0.1 is a workaround to this issue.

--- a/fsspec/tests/test_parquet.py
+++ b/fsspec/tests/test_parquet.py
@@ -86,7 +86,7 @@ def test_open_parquet_file(
     with open(path, "wb") as f:
         f.write(b"0" * file_size)
 
-        if footer_sample_size == 8:
+        if footer_sample_size == 8 and columns is not None:
             # We know 8 bytes is too small to include
             # the footer metadata, so there should NOT
             # be a key for the last 8 bytes of the file

